### ISSUE TITLE
Netlify dev crashes when editing functions

### ISSUE THREAD

**Issue by timroes** [netlify/cli#2245]
**Describe the bug**

I assume this is related #2244

When editing function files and saving them, sometimes it happens that the full netlify dev process crashes when trying to call the API the next time:

```
◈ Function builder zip-it-and-ship-it building functions from directory /home/repo/src/server
◈ Function builder zip-it-and-ship-it finished building functions from directory /home/repo/src/server
[Error: ENOENT: no such file or directory, lstat '/home/repo/.netlify/functions-serve/exchange-rates.js'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'lstat',
  path: '/home/repo/.netlify/functions-serve/exchange-rates.js'
}
[Error: ENOENT: no such file or directory, lstat '/home/repo/.netlify/functions-serve/exchange-rates.js'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'lstat',
  path: '/home/repo/.netlify/functions-serve/exchange-rates.js'
}

/home/repo/node_modules/netlify-redirector/lib/redirects.js:116
      throw ex;
      ^
abort({"errno":-2,"code":"ENOENT","syscall":"lstat","path":"/home/repo/.netlify/functions-serve/exchange-rates.js"}) at Error
    at jsStackTrace (/home/repo/node_modules/netlify-redirector/lib/redirects.js:1070:13)
    at stackTrace (/home/repo/node_modules/netlify-redirector/lib/redirects.js:1087:12)
    at process.abort (/home/repo/node_modules/netlify-redirector/lib/redirects.js:8502:44)
    at process.emit (events.js:315:20)
    at processPromiseRejections (internal/process/promises.js:245:33)
    at processTicksAndRejections (internal/process/task_queues.js:94:32)
error Command failed with exit code 7.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

This happens way more rarely then the issue described in #2244 but when it happens the process dies but will leave a detached process for the original build tool (e.g. create react app) running, that needs to be killed manually via process manager/command line.

**To Reproduce**

I've unfortunately not found a reliable way to reproduce this, and hope this issue will be resolved automatically via #2244. Wanted to log it nevertheless in case someone else comes across this, and potentially has a reliable way reproducing this.

**Configuration**

Since the stack trace indicates this is related to redirects, the `netlify.toml` in my project where this happens has the following redirects:

```
# Make all netlify functions available under /api/*
[[redirects]]
  from ="/api/*"
  to = "/.netlify/functions/:splat"
  status = 200
  force = true

# SPA serve all URLs via index.html
[[redirects]]
  from = "/*"
  to = "/index.html"
  status = 200
```

Other than that it's pretty much the config shown in the reproduction repository in #2244 (https://github.com/timroes/netlify-issues/tree/issue/cli-rebuilding). Though I was never able reproducing it in that repository.

```bash
> npx envinfo --system --binaries --npmPackages netlify-cli --npmGlobalPackages netlify-cli

  System:
    OS: Linux 5.10 Arch Linux
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 13.64 GB / 31.01 GB
    Container: Yes
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.15.5 - ~/.nvm/versions/node/v14.15.5/bin/node
    Yarn: 1.22.10 - ~/repo/node_modules/.bin/yarn
    npm: 6.14.11 - ~/.nvm/versions/node/v14.15.5/bin/npm
  npmPackages:
    netlify-cli: ^3.21.12 => 3.21.12 
```

**Expected behavior**

The process should not crash after changing function files.

**Comment by eduardoboucas** [comment]
@timroes Thanks for the detailed report! Super helpful. Version `3.22.0` should fix this. Could you please try and let me know?

**Comment by timroes** [comment]
It seems this is fixed with 3.22.0, since this happened more rarely than #2244 not sure if I might just have been lucky, but I am going to close this as fixed for now, and would reopen if it happens again. Thanks a lot.
